homeassistant/components/music_assistant/__init__.py

```diff
@@ -97,7 +99,10 @@ async def on_hass_stop(event: Event) -> None:
         listen_task.cancel()
         raise ConfigEntryNotReady("Music Assistant client not ready") from err
 
+    # store the listen task and mass client in the entry data
     entry.runtime_data = MusicAssistantEntryData(mass, listen_task)
+    # register our custom services
+    register_services(hass, mass)
```

Comment:
This should happen in `async_setup`

Reply:
Yeah I was afraid you were going to say that but then I look the ref to the client and need to do a lookup (for the first entry found)

Reply:
Yes, that's what we want

Reply:
Take a look at `mealie` for a nice example. It checks both if the config entry exists and if its loaded, if its not, it wil raise an error

Reply:
yeah, I have something like this. Just waiting for the bump PR to be finished so I can rebase this one